src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MonoDroidMarkStep.cs

```diff
@@ -211,15 +214,25 @@ protected override void DoAdditionalMethodProcessing (MethodDefinition method)
 		{
 			string member, nativeMethod, signature;
 
-			if (!method.TryGetRegisterMember (out member, out nativeMethod, out signature))
-				return;
+			bool preserveJniMarshalMethodOnly = false;
+			if (!method.TryGetRegisterMember (out member, out nativeMethod, out signature)) {
+				if (PreserveJniMarshalMethods () &&
+				    method.DeclaringType.GetMarshalMethodsType () != null &&
+				    method.TryGetBaseOrInterfaceRegisterMember (out member, out nativeMethod, out signature))
```

Comment:
personal preference. Given the number of & params it might be easier to read if we have `{` `}` for the code block.